### PR TITLE
Support for non-OpenAI models

### PR DESCRIPTION
**Pull request by alexanderatallah**
Love Context - fantastic idea!

This is just a suggestion PR in case you want to add support for hosted, non-OpenAI models (https://openrouter.ai), including CodeLlama. It also enables free (er, 100% discounted) inference, if you want to use Mistral 7B or Zephyr 7B.

Making this PR in part so I can use context with my openrouter models, and in part to hear your feedback

### CODE REVIEW

**Comment by jamino30** [comment]
Also, don't forget to mention in the `README.md` that their `OPENAI_API_KEY` is being used and show how to set it up correctly to avoid errors.

File: cli.py [hunk 1]
```diff
@@ -83,28 +83,31 @@ def main():
...
+        api_key = getpass("API key: ")
         openai.api_key = api_key
 
         print_markdown(
             """
         ---
 
         **Tip**: To save this key for later, run `export OPENAI_API_KEY=<your key>` on mac/linux or `setx OPENAI_API_KEY <your key>` on windows.
+        
+        For non-OpenAI models, you should set `OPENROUTER_API_KEY`, and optionally `OPENROUTER_APP_URL` and `OPENROUTER_APP_TITLE`.
```
Comment (adrwz) [inline cli.py:105]: Can you link to some instructions on how to use (where to get the API key/app url/app title)?

Ideally to the right sections in our README

**Review by adrwz — CHANGES_REQUESTED** [review]
This code looks great! Can you do a few things?

1. We've pushed a few changes today. Can you resolve merge conflicts & test again?
2. Can you add instructions on how to use OpenRouter to the README? Ideally right above the "Local Models" section

Cheers, & thank you for contributing!

File: cli.py (hunk 1 above)
Comment (alexanderatallah) [reply cli.py:105]: Will do. FYI, if people don't set these, you'll get ranked for tokens consumed on the openrouter homepage :)

Comment (alexanderatallah) [reply cli.py:105]: If a developer does set a referrer app url, then the developer will get ranked

**Comment by alexanderatallah** [comment]
@adrwz Resolved your comments!

**Comment by adrwz** [comment]
@alexanderatallah tested & made some changes, LMK what you think!

https://github.com/alexanderatallah/context/pull/1

**Review by adrwz — CHANGES_REQUESTED** [review]
@alexanderatallah tested & made some changes to your PR, LMK what you think!

- Calling openai.chatCompletion.create() no longer works w/ most recent openai package version (see https://github.com/fleet-ai/context/issues/1), and openai.chat.completions.create() is not compatible with OpenRouter, so I changed it to curl the URL
- We'll only ask for OpenRouter API keys if the user asks to use a non-OpenAI model

https://github.com/alexanderatallah/context/pull/1

**Comment by alexanderatallah** [comment]
Just saw @adrwz - approving now

**Comment by adrwz** [comment]
Merged! Ty for contributing 🙏

**Pull request merged**